### DISCUSSION TITLE
GPS Stations as "Tie Points" instead of Reference Points

### DISCUSSION THREAD

**Discussion by rbiessel** [insarlab/MintPy#804] [Ideas]
I'm not sure I really understand the reference point selection when displaying a velocity map. From what I understand, the velocities are referenced to some point, as opposed to re-referencing the time series and recomputing the velocity map? It also seems then that `--ref-gps` option sets the velocity at that point to zero, regardless of the actual velocity of that station.

 I would like to be able to reference the initial time-series to a GPS station and I would like to be able to use it as a tie point rather than simply a reference point. In other words, I want to be able reproject GPS station components in the same way (enu2los or up2los) and then adjust the InSAR time-series so it agrees with this GPS station. (Going even further, it would be great to be able to either interpolate tie-points between different GPS stations or fit a least-squares plane to reference the InSAR timeseries to.)

In other words, I would like to be able to fix the InSAR time-series to some GPS time-series instead of simply using a GPS station velocity as a reference. 

For environments with very spatially heterogeneous surface movements where it's find any pixel that is actually stable, this would be quite useful.

I want to get around to implementing this into MintPy myself, but I thought I'd suggest it here if anyone is planning on doing something similar already in the meantime.

**Comment by welcome** [comment]
👋  Thanks for opening your first issue here! Please make sure you filled out the template with as much detail as possible. We appreciate that you took the time to contribute!
Make sure you read our [contributing guidelines](https://github.com/insarlab/MintPy/blob/master/docs/CONTRIBUTING.md)

**Comment by yunjunz** [comment]
1. re-referencing the velocity from point P to Q
2. re-referencing the time-series from point P to Q, then re-calculate the linear velocity

The resulting linear velocities are exactly the same between 1 and 2. @olliestephenson and I confirmed this via math and real data experiment.

> I'm not sure I really understand the reference point selection when displaying a velocity map. From what I understand, the velocities are referenced to some point, as opposed to re-referencing the time series and recomputing the velocity map? It also seems then that `--ref-gps` option sets the velocity at that point to zero, regardless of the actual velocity of that station.

Your understanding is correct.

**Comment by yunjunz** [comment]
@rbiessel Your idea of tieing the InSAR time-series or velocity to GNSS via one or multiple tie points sounds all good to me. I am happy to answer any questions related to MintPy that you may have.

You may want to implement it in a jupyter notebook or eventually an independent script, I would not suggest integrating it into `view.py` as it's pretty complicated there already.
